saltproc/tests/test_simulation.py

```diff
@@ -1,41 +1,440 @@
+
+        assert tpois_after_params[0] == pois_after.mass
+        assert tpois_after_params[1] == pois_after.density
+        assert tpois_after_params[2] == pois_after.vol
+        assert tpois_after_params[3] == pois_after.temp
+        assert tpois_after_params[4] == pois_after.mass_flowrate
+        assert tpois_after_params[5] == pois_after.void_frac
+        assert tpois_after_params[6] == pois_after.burnup
+    except AssertionError:
+        db.close()
+        os.remove(db_file)
+        simulation.db_path = db_path_old
+        raise AssertionError("incorrect Value")
+    except KeyError:
+        db.close()
+        os.remove(db_file)
+        simulation.db_path = db_path_old
+        raise KeyError("incorrect key")
+    except BaseException:
+        db.close()
+        os.remove(db_file)
+        simulation.db_path = db_path_old
+        print("something went wrong. See error stack for details")
+
+    # close the file
+    db.close()
+
+    # delete test file
+    os.remove(db_file)
+
+    # use original db path
+    simulation.db_path = db_path_old
+
+
+def test_store_run_init_info():
+    """
+    This unit test checks that the entries ``store_run_init_info()`
+    stores in the database match the corresponding entries from the input
+    explicity in value and implicitly in type.
+    """
+
+    # read data
+    simulation.sim_depcode.read_depcode_info()
+    init_info = simulation.sim_depcode.sim_info
+
+    # we want to keep the old path for other sims, but for this
+    # test we'll want a fresh db
+    db_path_old = simulation.db_path
+    db_file = serpent.iter_inputfile + '.h5'
+    simulation.db_path = db_file
+
+    # store data at
+    simulation.store_run_init_info()
+
+    # read stored data
+    try:
+        db = tb.open_file(simulation.db_path, mode='r',
+                          filters=simulation.compression_params)
+    except Exception:
+        db.close()
+        print('Unable to assign correct value to db.\
+              See error stack for more info.')
+
+    tinit_info = db.root.initial_depcode_siminfo[0]
+
+    try:
+        assert tinit_info[0] == simulation.sim_depcode.npop
+        assert tinit_info[1] == simulation.sim_depcode.active_cycles
+        assert tinit_info[2] == simulation.sim_depcode.inactive_cycles
+        assert str(tinit_info[3])[2:-1] == init_info['depcode_name']
+        assert str(tinit_info[4])[2:-1] == init_info['depcode_version']
+        assert str(tinit_info[5])[2:-1] == init_info['title']
+        assert str(tinit_info[6])[2:-1] == init_info['depcode_input_filename']
+        assert str(tinit_info[7])[2:-1] == init_info['depcode_working_dir']
+        assert str(tinit_info[8])[2:-1] == init_info['xs_data_path']
+        assert tinit_info[9] == init_info['OMP_threads']
+        assert tinit_info[10] == init_info['MPI_tasks']
+        assert tinit_info[11] == init_info['memory_optimization_mode']
+        assert tinit_info[12] == init_info['depletion_timestep']
```

Comment:
Similar comment to above re:input file line order

Reply:
This case is a bit different. The function `store_run_init_info` stores all of those parameters in a new datatype, `sim_info_dtype`:
```
sim_info_dtype = np.dtype([
             ('neutron_population', int),
             ('active_cycles', int),
             ('inactive_cycles', int),
             ('depcode_name', 'S20'),
             ('depcode_version', 'S20'),
             ('title', 'S90'),
             ('depcode_input_filename', 'S90'),
             ('depcode_working_dir', 'S90'),
             ('xs_data_path', 'S90'),
             ('OMP_threads', int),
             ('MPI_tasks', int),
             ('memory_optimization_mode', int),
             ('depletion_timestep', float)
         ])
```
The relevant variables are then stored in the appropriate fields of a variable of this datatype. This variables is then stored in the database.

If you explore the database manually using `pytables` you'll see this:
```
/initial_depcode_siminfo (Table(1,), fletcher32, shuffle, blosc(9)) 'Initial depletion code simulation parameters'
  description := {
  "neutron_population": Int64Col(shape=(), dflt=0, pos=0),
  "active_cycles": Int64Col(shape=(), dflt=0, pos=1),
  "inactive_cycles": Int64Col(shape=(), dflt=0, pos=2),
  "serpent_version": StringCol(itemsize=20, shape=(), dflt=b'', pos=3),
  "title": StringCol(itemsize=90, shape=(), dflt=b'', pos=4),
  "serpent_input_filename": StringCol(itemsize=90, shape=(), dflt=b'', pos=5),
  "serpent_working_dir": StringCol(itemsize=90, shape=(), dflt=b'', pos=6),
  "xs_data_path": StringCol(itemsize=90, shape=(), dflt=b'', pos=7),
  "OMP_threads": Int64Col(shape=(), dflt=0, pos=8),
  "MPI_tasks": Int64Col(shape=(), dflt=0, pos=9),
  "memory_optimization_mode": Int64Col(shape=(), dflt=0, pos=10),
  "depletion_timestep": Float64Col(shape=(), dflt=0.0, pos=11)}
  byteorder := 'little'
  chunkshape := (150,)
```
So it is much easier for us to determine if something got changed if a test doesn't pass. Since this part of the database is fairly descriptive about its data entries, I don't think it's absolutley necessary to change the way the data gets stored, but again it would be nice to directly query for this info without knowing about the index order.